### PR TITLE
Add explicit name attribute to all packages

### PR DESCRIPTION
**Pull request by timhoffm**
This:

- save us from having to figure out the project name from a repository location, which will become more complicated when we accept non-GitHub repos (#100).
- allows using whitespace and arbitrary capitalization in the project name (which might be spelled differently than the repository name).

### CODE REVIEW

**Comment by jklymak** [comment]
This is fine, though it seems redundant for 99% of packages.  I would make it optional for the packages that want a name different than the repo or don't have a repo.

**Comment by timhoffm** [comment]
> Can you please also update the readme template?

@story645 What do you miss?

**Comment by timhoffm** [comment]
> This is fine, though it seems redundant for 99% of packages. I would make it optional for the packages that want a name different than the repo or don't have a repo.

IMHO there is value in making this mandatory. With a mandatory name, we can get rid of any repo parsing. Having distinct unrelated attributes is both simpler in code and simpler to explain to the end user (Understanding "Name is optional, if not given we extract it from `repo`" is much more effort than typing your project name).

File: packages/DnaFeaturesViewer.yml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+name: DNA Feature Viewer
```
Comment (QuLogic) [inline packages/DnaFeaturesViewer.yml:1]:
```suggestion
name: DNA Features Viewer
```

**Review by QuLogic — APPROVED** [review]

**Pull request merged**